scripts/zkai_d128_block_receipt_composition_gate.py

```
@@ -0,0 +1,1123 @@
+    if require_mutations:
+        cases = payload.get("cases")
+        if not isinstance(cases, list):
+            raise D128BlockReceiptError("mutation cases must be a list")
+        expected = expected_mutation_inventory()
+        expect_equal(payload.get("case_count"), len(cases), "case count length")
+        expect_equal(payload.get("case_count"), len(expected), "case count inventory")
+        expect_equal(summary.get("mutation_cases"), len(cases), "summary mutation count")
+        rejected_count = 0
+        for index, (case, expected_case) in enumerate(zip(cases, expected, strict=True)):
+            if not isinstance(case, dict):
+                raise D128BlockReceiptError("mutation case must be an object")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)